tests/src/Functional/DeveloperAppFieldTest.php

```diff
@@ -87,6 +88,161 @@ protected function tearDown() {
+        'options' => [],
+        'attributes' => [],
+      ],
+    ];
+
+    $fields = [
+      strtolower($this->randomMachineName()) => [
+        'type' => 'boolean',
+        'data' => [
+          ['value' => TRUE],
+          ['value' => FALSE],
+        ],
+        'encoded' => '1,',
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'float',
+        'data' => [
+          ['value' => M_PI],
+          ['value' => M_E],
+          ['value' => M_EULER],
+        ],
+        'encoded' => implode(',', [M_PI, M_E, M_EULER]),
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'integer',
+        'data' => [
+          ['value' => 4],
+          ['value' => 9],
+        ],
+        'encoded' => '4,9',
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'decimal',
+        'data' => [
+          ['value' => '0.1'],
+        ],
+        'encoded' => '0.1',
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'list_float',
+        'settings' => [
+          'settings[allowed_values]' => implode(PHP_EOL, [
+            M_PI,
+            M_E,
+            M_EULER,
+          ]),
+        ],
+        'data' => [
+          ['value' => M_PI],
+        ],
+        'encoded' => (string) M_PI,
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'list_integer',
+        'settings' => [
+          'settings[allowed_values]' => implode(PHP_EOL, [1, 2, 3]),
+        ],
+        'data' => [
+          ['value' => 2],
+          ['value' => 3],
+        ],
+        'encoded' => '2,3',
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'list_string',
+        'settings' => [
+          'settings[allowed_values]' => implode(PHP_EOL, [
+            'qwer',
+            'asdf',
+            'zxcv',
+          ]),
+        ],
+        'data' => [
+          ['value' => 'qwer'],
+          ['value' => 'asdf'],
+          ['value' => 'zxcv'],
+        ],
+        'encoded' => 'qwer,asdf,zxcv',
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'string',
+        'data' => [
+          ['value' => $paragraph],
+        ],
+        'encoded' => "\"{$paragraph}\"",
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'string_long',
+        'data' => [
+          ['value' => $paragraphs],
+        ],
+        'encoded' => "\"{$paragraphs}\"",
+      ],
+      strtolower($this->randomMachineName()) => [
+        'type' => 'link',
+        'data' => $link,
+        'encoded' => json_encode($link),
+      ],
+    ];
+
+    foreach ($fields as $name => $data) {
+      $this->fieldUIAddNewField(
+        '/admin/config/apigee-edge/app-settings',
+        $name, strtoupper($name),
+        $data['type'],
+        ($data['settings'] ?? []) + [
+          'cardinality' => -1,
+        ],
+        []
+      );
+
+      drupal_flush_all_caches();
```

Comment:
Just a comment: As I have already told we should figure out and document why a `drupal_flush_all_caches()` is needed in a test. We should not rely on cache clears in tests because those tests my provide false positive results and hide an underlying issue in the system that will be hard to find with all green tests.